assets/js/modules/analytics-4/datastore/custom-dimensions.test.js

```diff
@@ -283,6 +264,37 @@ describe( 'modules/analytics-4 custom-dimensions', () => {
 				).toEqual( customDimensionNames );
 			} );
 		} );
+
+		describe( 'receiveCreateCustomDimensionError', () => {
+			it( 'sets error in the datastore for the provided custom dimension', () => {
+				registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
+					propertyID,
+				} );
+
+				const error = {
+					code: 'test-error-code',
+					message: 'Test error message',
+					data: {
+						reason: 'test-error-reason',
+					},
+				};
+
+				registry
+					.dispatch( MODULES_ANALYTICS_4 )
+					.receiveCreateCustomDimensionError(
+						error,
+						'googlesitekit_post_date'
+					);
+
+				expect(
+					registry
+						.select( MODULES_ANALYTICS_4 )
+						.getCreateCustomDimensionError(
+							'googlesitekit_post_date'
+						)
```

Comment:
Understood, thank you @aaemnnosttv. I only used the selector here because it was a little inconvenient to access the error in the store because we need to generate the error key using the custom dimension options, which we don't have access to as it is not exported from `custom-dimensions.js`.

However, I've worked around that by adding a local copy of the custom dimension options in the test case and generating the error copy there.

Let me know what you think, thanks!